elasticsearch/templates/statefulset.yaml

```diff
@@ -108,6 +108,20 @@ spec:
       - name: "{{ template "name" . }}"
         image: "{{ .Values.image }}:{{ .Values.imageTag }}"
         imagePullPolicy: "{{ .Values.imagePullPolicy }}"
+        {{- if eq .Values.roles.master "true" }}
+        command:
+          - /bin/sh
+          - -c
+          - |
+            # https://github.com/elastic/helm-charts/issues/63#issuecomment-471096950
+            # Run ES as a background task, and forward SIGTERM to it, then wait for it to exit
+            trap 'kill $(jobs -p)' SIGTERM
+            /usr/local/bin/docker-entrypoint.sh elasticsearch &
+            wait
+            # now keep the pod alive for 30s after ES dies so that we will refuse connections from
+            # the new master rather than them needing to time  out
+            sleep 30
```

Comment:
Why 30 seconds? How do you know that this will be enough time? Magic numbers are not normally a good sign and should only be used if there is no way to determine how long we should wait. 

Luckily it is possible to detect that a new master has been elected as I commented in https://github.com/elastic/helm-charts/issues/63#issuecomment-465081133. This is what I would prefer to see implemented. As it doesn't rely on magic numbers and it also means that we can continue to use the default entrypoint for the docker images.

Reply:
added `sleepSecondsOnTermination` value and set `30` as default value.

Reply:
Making it configurable is certainly better than hard coding it. But this doesn't address my original feedback from https://github.com/elastic/helm-charts/pull/82#discussion_r266901909. Why is this 30 seconds, and why is that long enough to prevent the problem defined in #63. Is this calculated based on something?

Reply:
I just copied and pasted https://github.com/elastic/helm-charts/issues/63#issuecomment-471096950 . That's where 30 seconds came from.  In my test environment, around 10 seconds were sufficient.